diff --git a/core/modules/media/media.permissions.yml b/core/modules/media/media.permissions.yml
index 5c5692c..64ebf58 100644
--- a/core/modules/media/media.permissions.yml
+++ b/core/modules/media/media.permissions.yml
@@ -33,5 +33,8 @@ access media overview:
   title: 'Access media overview'
   description: 'Users with this permission can access the media overview page.'
 
+view media types:
+  title: 'View media types'
+
 permission_callbacks:
   - \Drupal\media\MediaPermissions::mediaTypePermissions
diff --git a/core/modules/media/src/MediaTypeAccessControlHandler.php b/core/modules/media/src/MediaTypeAccessControlHandler.php
index 2f134a6..e850b9d 100644
--- a/core/modules/media/src/MediaTypeAccessControlHandler.php
+++ b/core/modules/media/src/MediaTypeAccessControlHandler.php
@@ -26,6 +26,9 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
     if ($operation === 'view label') {
       return AccessResult::allowedIfHasPermission($account, 'view media');
     }
+    elseif ($operation === 'view') {
+      return AccessResult::allowedIfHasPermission($account, 'view media type');
+    }
     else {
       return parent::checkAccess($entity, $operation, $account);
     }
