diff --git a/core/modules/language/language.module b/core/modules/language/language.module
index c7873b8..4fb33eb 100644
--- a/core/modules/language/language.module
+++ b/core/modules/language/language.module
@@ -317,8 +317,10 @@ function language_modules_installed($modules) {
     /** @var \Drupal\language\LanguageNegotiatorInterface $negotiator */
     $negotiator = \Drupal::service('language_negotiator');
     $configurable = \Drupal::config('language.types')->get('configurable');
-    $negotiator->updateConfiguration($configurable);
-    $negotiator->purgeConfiguration();
+    if (is_array($configurable)) {
+      $negotiator->updateConfiguration($configurable);
+      $negotiator->purgeConfiguration();
+    }
   }
   else {
     // In language_entity_base_field_info_alter() we are altering view/form
